### PR TITLE
Make orderly_pull_archive parameter arg more behaviour more predictable

### PR DESCRIPTION
**Pull request by r-ash**
This PR will:
* Construct a default query from parameters if `id = "latest"`
* Update docs to hopefully make this behaviour clear

### CODE REVIEW

File: R/query_search.R [hunk 1]
```diff
@@ -450,3 +450,26 @@ remote_report_metadata <- function(name, remote, config) {
...
+    stop(e)
+  })
+  quote_character <- function(x) {
+    out <- x
+    if (is.character(x)) {
+      out <- paste0('"', x, '"')
+    }
+    out
+  }
+  params <- lapply(parameters, quote_character)
```
Comment (richfitz) [inline R/query_search.R:471]: I had suggested

```
  params <- sprintf(dquote(list_to_character(lapply(parameters, as.character)))
```

but realise that this is even easier:

```suggestion
  params <- sprintf("parameter:%s == %s", names(parameters), names(parameters))
```

and then just paste that together with ` && `.

The query interface should do the validation of the parameters for us, we should not do the conversion to character etc here

File: R/remote.R [hunk 2]
```diff
@@ -49,13 +49,19 @@
...
 ##'   resolution.  If you are using a query for \code{id} that
 ##'   involves a parameter (e.g., \code{latest(parameter:x == p)}) you
 ##'   will need to pass in the parameters here.  Similarly, if you are
 ##'   pulling a report that uses query dependencies that reference
 ##'   parameters you need to pass them here (the same parameter set
-##'   will be passed through to all dependencies).
+##'   will be passed through to all dependencies). If passed with
+##'   \code{id = "latest"} then query will be constructed to pull latest
+##'   version of report with report parameters equal to those passed e.g. if
+##'   \code{parameters = list(nmin = 0.5, nmax = 1)} this will pull latest
```
Comment (richfitz) [inline R/remote.R:61]: I'd write out here the implied query, something like

> this is equivalent to providing the query `latest(parameter:nmin = nmin && parameter:nmax:nmax)` along with your parameters

**Review by richfitz — CHANGES_REQUESTED** [review]
Couple of things that will simplify this, sorry forgot how this all works when we chatted as it's actually even simpler than we'd thought

File: R/query_search.R (hunk 1 above)
Comment (r-ash) [reply R/query_search.R:471]: ooh ofc yes

**Review by richfitz — APPROVED** [review]

**Pull request merged**